src/app/modules/myinfo/myinfo.service.ts

```diff
@@ -270,6 +280,61 @@ export class MyInfoServiceClass {
       const { fieldValue, isReadOnly } = myInfoData.getFieldValueForAttr(
         myInfoAttr as InternalAttr,
       )
+
+      // Add logging to check if Myinfo field value exists in our constants lists
+      const isFieldValueInMyinfoListCheck = (
+        fieldValue: string,
+        myInfoAttr: string | string[],
+        myInfoList: string[],
+      ) => {
+        const isFieldValueInMyinfoList = myInfoList.includes(fieldValue)
+        if (!isFieldValueInMyinfoList) {
+          logger.error({
+            message: 'Myinfo field value not found in existing list',
+            meta: {
+              action: 'prefillAndSaveMyInfoFields',
+              myInfoFieldValue: fieldValue,
+              myInfoAttr,
+            },
+          })
+        }
+      }
+
+      // Check if field value exists in our constants lists
+      if (fieldValue) {
+        if (myInfoAttr === MyInfoAttribute.Occupation) {
+          isFieldValueInMyinfoListCheck(
+            fieldValue,
+            myInfoAttr,
+            myInfoOccupations,
+          )
+        } else if (
+          myInfoAttr === MyInfoAttribute.Race ||
+          myInfoAttr === MyInfoAttribute.ChildRace ||
+          myInfoAttr === MyInfoAttribute.ChildSecondaryRace
+        ) {
+          isFieldValueInMyinfoListCheck(fieldValue, myInfoAttr, myInfoRaces)
+        } else if (myInfoAttr === MyInfoAttribute.Nationality) {
+          isFieldValueInMyinfoListCheck(
+            fieldValue,
+            myInfoAttr,
+            myInfoNationalities,
+          )
+        } else if (myInfoAttr === MyInfoAttribute.Dialect) {
+          isFieldValueInMyinfoListCheck(fieldValue, myInfoAttr, myInfoDialects)
+        } else if (myInfoAttr === MyInfoAttribute.BirthCountry) {
+          isFieldValueInMyinfoListCheck(fieldValue, myInfoAttr, myInfoCountries)
+        } else if (myInfoAttr === MyInfoAttribute.HousingType) {
+          isFieldValueInMyinfoListCheck(
+            fieldValue,
+            myInfoAttr,
+            myInfoHousingTypes,
+          )
+        } else if (myInfoAttr === MyInfoAttribute.HdbType) {
+          isFieldValueInMyinfoListCheck(fieldValue, myInfoAttr, myInfoHdbTypes)
+        }
+      }
```

Comment:
Should we log an error as a 'default' case?  Or use switch-cases?

Reply:
won't log errors in default cases, refactored here [83cb27a](https://github.com/opengovsg/FormSG/pull/6694/commits/83cb27af4977997df567bec80d98593cf73c1f54) :)